src/sentry/api/serializers/models/project.py

```diff
@@ -420,15 +406,48 @@ def get_attrs(self, item_list, user):
                 "dateFinished": date_finished,
             }
 
+        return deploys_by_project
+
+    def get_attrs(self, item_list, user):
+        attrs = super(ProjectSummarySerializer, self).get_attrs(item_list, user)
+
+        projects_with_user_reports = set(
+            UserReport.objects.filter(project_id__in=item_list).values_list("project", flat=True)
+        )
+
+        project_envs = (
+            EnvironmentProject.objects.filter(
+                project_id__in=[i.id for i in item_list],
+                # Including the organization_id is necessary for postgres to use indexes
+                # efficiently.
+                environment__organization_id=item_list[0].organization_id,
+            )
+            .exclude(
+                is_hidden=True,
+                # HACK(lb): avoiding the no environment value
+            )
+            .exclude(environment__name="")
+            .values("project_id", "environment__name")
+        )
+
+        environments_by_project = defaultdict(list)
+        for project_env in project_envs:
+            environments_by_project[project_env["project_id"]].append(
+                project_env["environment__name"]
+            )
+
         # We just return the version key here so that we cut down on response size
         latest_release_versions = {
             release.actual_project_id: {"version": release.version}
             for release in bulk_fetch_project_latest_releases(item_list)
         }
 
+        if not self._collapse("latestDeploys"):
+            deploys_by_project = self.get_deploys_by_project(item_list)
         for item in item_list:
             attrs[item]["latest_release"] = latest_release_versions.get(item.id)
-            attrs[item]["deploys"] = deploys_by_project.get(item.id)
+            if not self._collapse("latestDeploys"):
```

Comment:
Can you make "latestDeploys" a constant? Going to be going back and doing this on the others as well.